server/controllers/authController.api.js

```diff
@@ -1,39 +1,82 @@
 'use strict';
 const { User } = require('../models');
-const { encodeToken, mongoDbErrorHandler } = require('../utils');
+const { encodeToken, decodeToken, mongoDbErrorHandler } = require('../utils');
 
 exports.register = (req, res) => {
-  User.create(req.body, (err, user) => {
-    if (err) {
-      mongoDbErrorHandler(err, res);
-    } else {
-      const { name, email } = user;
-      const token = encodeToken({ name, email });
-      return res.status(201).json({ token });
-    }
-  });
-};
+    User.create(req.body, (err, user) => {
+        if (err) {
+            mongoDbErrorHandler(err, res);
+        } else {
+            const { name, email, _id } = user;
+            const token = encodeToken({ name, email, _id });
+            return res.status(201).json({
+                token,
+                user: {
+                    _id,
+                    name,
+                    email
+                }
+            });
+        }
+    });
+}
 
 exports.login = (req, res) => {
-  const { email, password } = req.body;
-  User.findOne({ email }, (err, user) => {
-    if (err) {
-      // Unknown/MongoDB error
-      mongoDbErrorHandler(err, res, 500);
-    } else if (user && user.comparePassword(password)) {
-      // Login OK
-      const { name, email } = user;
-      const token = encodeToken({ name, email });
+    const { name, password } = req.body;
+    User.findOne({ name })
+        .populate('projects')
+        .exec((err, user) => {
+            if (err) {
+                mongoDbErrorHandler(err, res, 500);
```

Comment:
I think Shums mentioned that we should not use 500 error codes actually, but I do know this was carried over from when the route was created

Reply:
Ah, I didn't see that one! What code should we use in this case?

Reply:
I think 400, but a 4xx error I guess? 500 is Internal Server Error

Reply:
Probably take a look at why it would fail and it'll give us a better sense of the error code - it may be because the user doesn't exist (in case 404), or because of a bad formatted name (400), etc. 

Sometimes people throw 500s, but it's really a catch all and should be avoided